portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -38,17 +38,12 @@ public void init() {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
-    // convert list of messages to JSON
-    String json = convertToJson(messages);
-
-    response.setContentType("application/json;"); // no longer text/html
+    response.setContentType("application/json;");
     response.getWriter().println(messages);
```

Comment:
mm, are messages supposed to be the converted json? otherwise your new function convertToJson is not used anywhere right?